File: teacherapp-web/src/main/web/src/app/dto/folder.ts
import { File } from './file';

export class Folder {
    private _id: number;
    private _name: string;
    private _files: File[] = [];

    public static createEmptyFolder() {
        return new Folder({
            id : '',
            name: '',
            files: []
        });
    }

    constructor(data: any) {
        this._id = data.id;
        this._name = data.name;
        data.files.forEach(file => {
            this._files.push(new File(file));
        });
    }

    get id(): number {
        return this._id;
    }

    set id(value: number) {
        this._id = value;
    }

    get name(): string {
        return this._name;
    }

    set name(value: string) {
        this._name = value;
    }

    get files(): File[] {
        return this._files;
    }

    set files(value: File[]) {
        this._files = value;
    }

    public toObject() {
        return {
            id: this.id,
            name: this.name,
            files: this.files,
        };
    }
}

File: teacherapp-server/src/main/java/com/kulygin/teacherapp/domain/Folder.java
package com.kulygin.teacherapp.domain;

import lombok.*;

import javax.persistence.*;
import java.util.List;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(schema = "public")
@EqualsAndHashCode(of = {"id"})
@ToString(exclude = {"files"})
public class Folder {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Integer id;
    private String name;
    @OneToMany(mappedBy="folder", fetch = FetchType.LAZY)
    private List<File> files;
}
